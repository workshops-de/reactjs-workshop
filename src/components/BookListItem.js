import React from 'react'
import { Link } from 'react-router-dom'
import BookShape from '../shapes/book'

const BookListItem = ({ data }) => {
  const { title, subtitle, author, isbn } = data
  return (
    <div>
      <div>
        <strong>
          <Link to={`/books/${isbn}`}>{title}</Link>
        </strong>
      </div>
      <div>{subtitle}</div>
      <div>
        <i>{author}</i>
      </div>
      <div>ISBN: {isbn}</div>
    </div>
  )
}

BookListItem.propTypes = {
  data: BookShape,
}

export default BookListItem
