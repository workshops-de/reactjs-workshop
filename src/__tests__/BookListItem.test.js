import React from 'react'
import ReactTestRenderer from 'react-test-renderer'
import { MemoryRouter as Router } from 'react-router-dom'
import BookListItem from '../components/BookListItem'

const book = {
  title: 'Design Patterns',
  subtitle: 'Elements of Reusable Object-Oriented Software',
  isbn: '978-0-20163-361-0',
  abstract:
    'Capturing a wealth of experience about the design of object-oriented software, four top-notch designers present a catalog of simple and succinct solutions to commonly occurring design problems. Previously undocumented, these 23 patterns allow designers to create more flexible, elegant, and ultimately reusable designs without having to rediscover the design solutions themselves.',
  numPages: 395,
  author: 'Erich Gamma / Richard Helm / Ralph E. Johnson / John Vlissides',
  publisher: {
    name: 'Addison-Wesley',
    url: 'http://www.addison-wesley.de/',
  },
  id: '978-0-20163-361-0',
}

describe('A BookListItem component', function() {
  it('renders a book item correctly', () => {
    const tree = ReactTestRenderer.create(
      <Router>
        <BookListItem data={book} />
      </Router>,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
