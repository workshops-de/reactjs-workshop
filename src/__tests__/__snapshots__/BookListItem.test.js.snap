// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A BookListItem component renders a book item correctly 1`] = `
<div>
  <div>
    <strong>
      <a
        href="/books/978-0-20163-361-0"
        onClick={[Function]}
      >
        Design Patterns
      </a>
    </strong>
  </div>
  <div>
    Elements of Reusable Object-Oriented Software
  </div>
  <div>
    <i>
      Erich Gamma / Richard Helm / Ralph E. Johnson / John Vlissides
    </i>
  </div>
  <div>
    ISBN: 
    978-0-20163-361-0
  </div>
</div>
`;
